mat73/__init__.py

```diff
@@ -291,6 +284,11 @@ def loadmat(filename, use_attrdict=False, only_include=None, verbose=True):
     assert os.path.isfile(filename), '{} does not exist'.format(filename)
     decoder = HDF5Decoder(verbose=verbose, use_attrdict=use_attrdict,
                           only_include=only_include)
+    
+    ext = os.path.splitext(filename)[1].lower()
+    if ext!='.mat':
+        logging.warning('Can only load MATLAB .mat file, this file type might '
+                        f'be unsupported: {filename}')
```

Comment:
There is a new way to manipulate paths, maybe use it?

```IPython
In [1]: from pathlib import Path

In [2]: p = Path('tests/testfile1.mat')

In [3]: p.stem
Out[3]: 'testfile1'

In [4]: p.suffix
Out[4]: '.mat'
```

Reply:
Ah yes, `pathlib`, true. I'm just so used to os.path and never made the effort to switch. It had some minor disadvantages as they're now `Path` objects and no longer strings.